modules/heat_conduction/src/materials/AnisoHeatConductionMaterial.C

```diff
@@ -10,137 +10,152 @@
-
-    _has_temp(isCoupled("temp")),
-    _temperature(_has_temp ? coupledValue("temp") : _zero),
-
-    _my_thermal_conductivity_x(
-        isParamValid("thermal_conductivity_x") ? getParam<Real>("thermal_conductivity_x") : -1),
-    _my_thermal_conductivity_y(
-        isParamValid("thermal_conductivity_y") ? getParam<Real>("thermal_conductivity_y") : -1),
-    _my_thermal_conductivity_z(
-        isParamValid("thermal_conductivity_z") ? getParam<Real>("thermal_conductivity_z") : -1),
-
-    _thermal_conductivity_x_pp(isParamValid("thermal_conductivity_x_pp")
-                                   ? &getPostprocessorValue("thermal_conductivity_x_pp")
-                                   : NULL),
-    _thermal_conductivity_y_pp(isParamValid("thermal_conductivity_y_pp")
-                                   ? &getPostprocessorValue("thermal_conductivity_y_pp")
-                                   : NULL),
-    _thermal_conductivity_z_pp(isParamValid("thermal_conductivity_z_pp")
-                                   ? &getPostprocessorValue("thermal_conductivity_z_pp")
-                                   : NULL),
-
-    _my_specific_heat(isParamValid("specific_heat") ? getParam<Real>("specific_heat") : 0),
-
-    _thermal_conductivity_x(&declareProperty<Real>("thermal_conductivity_x")),
-    _thermal_conductivity_x_dT(&declareProperty<Real>("thermal_conductivity_x_dT")),
-    _thermal_conductivity_y(isParamValid("thermal_conductivity_y") ||
-                                    isParamValid("thermal_conductivity_y_pp")
-                                ? &declareProperty<Real>("thermal_conductivity_y")
-                                : NULL),
-    _thermal_conductivity_y_dT(
-        _thermal_conductivity_y ? &declareProperty<Real>("thermal_conductivity_y_dT") : NULL),
-    _thermal_conductivity_z(isParamValid("thermal_conductivity_z") ||
-                                    isParamValid("thermal_conductivity_z_pp")
-                                ? &declareProperty<Real>("thermal_conductivity_z")
-                                : NULL),
-    _thermal_conductivity_z_dT(
-        _thermal_conductivity_z ? &declareProperty<Real>("thermal_conductivity_z_dT") : NULL),
-
-    _specific_heat(declareProperty<Real>("specific_heat")),
+template <bool is_ad>
+AnisoHeatConductionMaterialTempl<is_ad>::AnisoHeatConductionMaterialTempl(
+    const InputParameters & parameters)
+  : DerivativeMaterialInterface<Material>(parameters),
+    _ref_temp(getParam<Real>("reference_temperature")),
+
+    _has_temp(isCoupled("temperature")),
+    _T(coupledGenericValue<is_ad>("temperature")),
+    _T_var(coupled("temperature")),
+    _T_name(getVar("temperature", 0)->name()),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
+
+    _user_provided_thermal_conductivity(getParam<std::vector<Real>>("thermal_conductivity")),
+    _user_provided_specific_heat(isParamValid("specific_heat") ? getParam<Real>("specific_heat")
```

Comment:
use a default value for the parameter instead of manually assigning it with `isParamValid`

Reply:
Done